source/common/modules/markdown-editor/hooks/codeblock-classes.ts

```diff
@@ -29,8 +29,10 @@ const findCodeDebounced = debounce(findCode, 400, { leading: true })
  */
 export default function codeblockClassHook (cm: CodeMirror.Editor): void {
   cm.on('keyHandled', handleNewline)
-  cm.on('cursorActivity', findCodeDebounced)
-  cm.on('optionChange', findCodeDebounced)
+  // cm.on('cursorActivity', findCodeDebounced)
+  cm.on('cursorActivity', findCode)
+  // cm.on('optionChange', findCodeDebounced)
+  cm.on('optionChange', findCode)
```

Comment:
During testing the Vim mode issue, I found that removing the `optionChange` listener indeed made Vim mode work again. I didn't fully understand why we have to listen for the `optionChange` event in the first place?

Also, I believe it would be sufficient to just listen to the `change` event instead `cursorActivity`. Well, during _typing_ both would be triggered (if I'm not mistaken), but I don't see a reason why the `findCode` function should be fired when only _navigating_ with the cursor.

Finally, listening to the `swapDoc` event would make sure that the codeblocks are rendered when we open/switch to a different document. (Currently, the `optionChange` event covers listening to file open events, but when we remove that, `swapDoc` could take over that part)

I tested a bit using the `change` and `swapDoc` hooks, and so far I didn't see any issue with Vim mode (and the normal mode).

Reply:
`optionChange` was indeed unnecessary, so removed that hook.

With regard to `cursorActivity` you are right. Changing as such!

Also, your suggestion with `swapDoc` is very good, added that event as well!